js/timers.js

```diff
@@ -79,14 +230,53 @@ function togglePanel() {
 
 window.addEventListener("DOMContentLoaded", function() {
     document.getElementById("timerConfigButton").addEventListener('click', togglePanel);
+    document.getElementById("closeConfigButton").addEventListener('click', togglePanel);
     document.getElementById("timerConfig").addEventListener("transitionend", hideConfigElements, true);
     var add_buttons = document.querySelectorAll("[data-action='add_timer']");
     for (i=0; i<add_buttons.length; ++i) {
         add_buttons[i].addEventListener("click", addTimer);
     }
     var clear_timers = document.querySelectorAll("[data-action='clear_timers']");
-    for (i=0; i<add_buttons.length; ++i) {
+    for (i=0; i<clear_timers.length; ++i) {
         clear_timers[i].addEventListener("click", removeAllTimers);
     }
 
-});
\ No newline at end of file
+});
+function startTimer(duration, display) {
+    var start = Date.now(),
+        diff,
+        minutes,
+        seconds;
+    function timer() {
+        // get the number of seconds that have elapsed since
+        // startTimer() was called
+        diff = duration - (((Date.now() - start) / 1000) | 0);
+
+        // does the same job as parseInt truncates the float
+        minutes = (diff / 60) | 0;
+        seconds = (diff % 60) | 0;
+
+        minutes = minutes < 10 ? "0" + minutes : minutes;
+        seconds = seconds < 10 ? "0" + seconds : seconds;
+
+        display.textContent = minutes + ":" + seconds;
+        if(minutes == "00" && seconds =="00"){
+            clearInterval(timerMap[display.id]);
+            display.parentNode.dataset.state="completed";
+        }
+        if (diff <= 0) {
+            // add one second so that the count down starts at the full duration
+            // example 05:00 not 04:59
+            start = Date.now() + 1000;
+        }
+    };
+    // we don't want to wait a full second before the timer starts
+    if(timerMap[display.id]!=null) {
+        clearInterval(timerMap[display.id]);
+    }
+    //timerMap[display.id]=timer;
+    //timerMap[display.id]();
+    timerMap[display.id]=setInterval(function(){timer()}, 1000);
```

Comment:
you can just use `setInterval(timer, 1000);` (to pass a function pointer like you would in c++, you just use the name without the ())